index.js

```diff
@@ -1,2 +1,601 @@
-// i has a cheezburger
-module.exports = require('./lib/speakeasy');
+'use strict';
+
+var base32 = require('base32.js');
+var crypto = require('crypto');
+var url = require('url');
+var util = require('util');
+
+/**
+ * Digest the one-time passcode options.
+ *
+ * @param {Object} options
+ * @param {String} options.secret Shared secret key
+ * @param {Integer} options.counter Counter value
+ * @param {String} [options.encoding="ascii"] Key encoding (ascii, hex,
+ *   base32, base64).
+ * @param {String} [options.algorithm="sha1"] Hash algorithm (sha1, sha256,
+ *   sha512).
+ * @param {String} [options.key] (DEPRECATED. Use `secret` instead.)
+ *   Shared secret key
+ * @return {Buffer} The one-time passcode as a buffer.
+ */
+
+exports.digest = function digest (options) {
+  var i;
+
+  // unpack options
+  var key = options.secret;
+  var counter = options.counter;
+  var encoding = options.encoding || 'ascii';
+  var algorithm = (options.algorithm || 'sha1').toLowerCase();
+
+  // Backwards compatibility - deprecated
+  if (options.key) {
+    console.log('Speakeasy - Deprecation Notice - Specifying the secret using `key` is no longer supported. Use `secret` instead.');
+    key = options.key;
+  }
+
+  // convert key to buffer
+  if (!Buffer.isBuffer(key)) {
+    key = encoding === 'base32' ? base32.decode(key)
+      : new Buffer(key, encoding);
+  }
+
+  // create an buffer from the counter
+  var buf = new Buffer(8);
+  var tmp = counter;
+  for (i = 0; i < 8; i++) {
+    // mask 0xff over number to get last 8
+    buf[7 - i] = tmp & 0xff;
+
+    // shift 8 and get ready to loop over the next batch of 8
+    tmp = tmp >> 8;
+  }
+
+  // init hmac with the key
+  var hmac = crypto.createHmac(algorithm, key);
+
+  // update hmac with the counter
+  hmac.update(buf);
```

Comment:
None of these implementations repeat the key:
- https://github.com/google/google-authenticator/blob/master/mobile/blackberry/src/com/google/authenticator/blackberry/PasscodeGenerator.java#L81
- https://github.com/gokyle/hotp/blob/master/hotp.go#L238
- https://github.com/google/google-authenticator/blob/master/mobile/ios/Classes/OTPGenerator.m#L102
- https://github.com/Jakobo/hotp-php/blob/master/hotp.php#L18
- https://github.com/adulau/hotp-js/blob/master/hotp/hotp.js#L89

Though I found a reference to padding for IPSec:
- http://www.rfc-archive.org/getrfc.php?rfc=4868